### PR TITLE
Feat(v5): add api auth package

### PR DESCRIPTION
**Pull request by SamSalvatico**
### Description

Added api-auth package

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [x] **New feature**
- [ ] **Dev change**
- [ ] **Additional tests**
- [ ] **Documentation**
- [ ] **Other**

### Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] This change might impact the developer experience of others and should be communicated

### CODE REVIEW

File: packages/api-auth/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+    "extends": "../../tsconfig.base.json",
+    "compilerOptions": {
+      "composite": true,
+      "outDir": "dist",
+    },
+    "include": ["src", "__tests__"],
+    "references": [
+      { "path": "../shared-errors" }
+    ]
+  }
```
Comment (nnorbert) [inline packages/api-auth/tsconfig.json:11]: Minor issue, EOF new line is missing.

**Review by nnorbert — APPROVED** [review]
LGTM!

File: packages/api-auth/package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  "name": "@ogcio/api-auth",
+  "version": "5.1.0",
+  "main": "dist/index.js",
+  "type": "module",
+  "dependencies": {
+    "@aws-sdk/client-kms": "^3.709.0",
+    "@fastify/sensible": "6.0.1",
+    "@ogcio/shared-errors": "1.0.0",
+    "fastify": "^5.1.0",
+    "fastify-plugin": "^4.5.1",
```
Comment (andreacappadona17) [inline packages/api-auth/package.json:11]:
```suggestion
    "fastify-plugin": "^5.0.1",
```

Comment (SamSalvatico) [reply packages/api-auth/package.json]: thank you @andreacappadona17

**Pull request merged**